mmengine/strategy/native_strategy.py

```diff
@@ -170,39 +169,47 @@ def save_checkpoint(self,
             callback(checkpoint)
 
         filepath = osp.join(out_dir, name)
+        self._save_checkpoint_to_file(checkpoint, filepath)
+
+    def _get_state_dict(self):
+        from mmengine.runner.checkpoint import get_state_dict, weights_to_cpu
+        return weights_to_cpu(get_state_dict(self.model))
+
+    def _get_optim_state_dict(self):
+        return self.optim.state_dict()
+
+    def _save_checkpoint_to_file(self, checkpoint, filepath):
+        from mmengine.runner.checkpoint import save_checkpoint
         save_checkpoint(checkpoint, filepath)
 
     def load_checkpoint(self,
                         load_dir: str,
-                        name: Optional[str] = None,
+                        filepath: Optional[str] = None,
                         load_optimizer: bool = False,
                         load_param_scheduler: bool = False,
                         *,
                         strict: bool = False,
                         map_location: Union[str, Callable] = 'cpu',
                         callback: Optional[Callable] = None) -> Optional[Dict]:
         # Avoid circular import
-        from mmengine.runner.checkpoint import (_load_checkpoint,
-                                                _load_checkpoint_to_model)
-        if name is None:
+        if filepath is None:
             # should auto detect latest checkpoint
-            name = self._latest_checkpoint_name(load_dir)
-        if name is None:
+            filepath = self._latest_checkpoint_name(load_dir)
+        if filepath is None:
             return None
```

Comment:
I have to clarify why this API seems kind of `weired`. It is designed to be compatible with [DeepSpeed](https://github.com/microsoft/DeepSpeed/blob/f1e4fb0b878e48d6fe33dbb2e0969d791d698e86/deepspeed/runtime/engine.py#LL3084C43-L3084C43) ones

```python
def save_checkpoint(self, save_dir, tag=None, client_state={}, save_latest=True):
    ....
```

1. It doesn't split `get_state_dict` and `save_checkpoint_to_file` into 2 stages, so you cannot get a state_dict first and then save it with your own method.
2. It saves the `tag` into a different file (`last_ckpt`, if I remembered well) than ours (`latest_checkpoint`, if I remembered well), and the `tag` is a relative path rather than a full path (that's what we did)
3. `tag` actually indicates a directory rather than a file.

For example, if you set `save_dir='aaa'` and `tag='bbb'` and `save_latest=True`, you will get something like:

```text
aaa
|- bbb
  |- model_rank_00.pth.
  |- model_rank_01.pth
  |- optim_rank_00.pth
  |- optim_rank_01.pth
|- last_ckpt  # a text file, where a single line of string 'bbb' is stored
```

And `load_checkpoint` is similar.